hw6/task1/library.js

```diff
@@ -0,0 +1,224 @@
+        if(myElement.constructor.name === "Track"){
+            this.Track.forEach(element => {
+                if(element === myElement){
+                    element[`${argument}`] = value;
+                }
+            });            
+        }
+                    //ALBUM
+        if(myElement.constructor.name === "Album"){
+            this.Album.forEach(element => {
+                if(element === myElement){
+                    element[`${argument}`] = value;
+                }
+            });            
+        }
+        if(myElement.constructor.name === "Artist"){
+            this.Artist.forEach(element => {
+                if(element === myElement){
+                    element[`${argument}`] = value;
+                }
+            });            
+        }        
+    }
+    
+//PLAY METHOD
+    play(){ 
+        const EventEmitter = require('events');
+        const player = new EventEmitter();              
+        let current = 0; 
+        let myTracks = this.Track;
+        let strInfo = `${myTracks[current].name}, Duration: ${myTracks[current].seconds}`; 
+        //EventEmitter next()        
+        this.nextTrack = function(){player.once('next', ()=>{
+            if(current + 1 >= myTracks.length){ 
+                current = 0;                
+            }
+            else{
+                current++;                            
+            }   
+                       
+        })};
+        //EventEmitter prev()    
+        this.prevTrack = function(){player.once('prev', ()=>{
+            if(current - 1 < 0){ 
+                console.log("Invalid command");               
+            }
+            else{
+                current--;                            
+            }   
+                       
+        })};
+        //EventEmitter pause()
+        this.pauseTrack = function(){player.once('pause', ()=>{
+            mainInterval.unref();
+            clearInterval(mainInterval);
+            console.log("pause");                        
+        })};
+
+        let mainInterval= setInterval(function(){
+            player.emit('next');
+            player.emit('prev');
+            player.emit('pause');                                                                                      
+            console.log(`${myTracks[current+1].name}, Duration: ${myTracks[current+1].seconds}`);            
+            current++;
+            if(current + 1 >= myTracks.length){
+                current = 0;
+                clearInterval(mainInterval);
+            }                                     
+        }, myTracks[current].seconds*1000);
+        //THIS CODE EMMITATION FOR WORK PAUSE AND PLAY        
+        mainInterval.ref();        
+        //END EMITATION
+        console.log("Play");
+        return strInfo;
+    };
+    next(){
+        this.nextTrack();
+    };
+    prev(){
+        this.prevTrack();
+    };
+    pause(){
+        this.pauseTrack();
+    };
+    search(value){
```

Comment:
search should work in entities that already has in Library class. that means - you can iterate evety of your arrays([see declaration line](https://github.com/Nikolaenko94/Auto-JS-2021/pull/6/files#diff-0ed900fd14b64e674fdf8e7dc523034d9c63dcebe14c33dd2981aeca1c5f266eR4)). and create a new array for finded entities. [Array.prototype.filter](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Array/filter) method can helps you